thriftpy2/parser/parser.py

```diff
@@ -404,6 +409,16 @@ def set_info(self, info):
         return self.index + 1
 
 
+class CurrentFieldSeqImplicitId(dict):
+    def __init__(self, *args, **kwargs):
+        self.index = -1
+
+    def set_info(self, info):
+        self[self.index] = info
```

Comment:
Why store the *info* in this object? Seems it doesn't accessed by any other place.

Reply:
This was also modeled off the `CurrentIncompleteType`, which stored the `info`. It wasn't 100% clear to me why that class also stored the info, but given I am unfamiliar with this library, ply, and lex and yacc in general, I just followed the existing pattern.

 I can see if I can remove the storage of the info here, and just use the class to track the index.